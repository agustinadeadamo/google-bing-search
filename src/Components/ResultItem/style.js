/**
 * @desc Dependencies
 */
import styled from "styled-components";

/**
 * @desc Variables
 */
import { Colors } from 'Mainstyles/Variables';

export const ResultItemComponent = styled.div`
    margin: 20px;

    .link {
        font-size: 12px;
        margin-bottom: 5px;
    }

    .title {
        font-size: 18px;
        color: ${Colors.black};
    }

    .snippet {
        font-size: 14px;
    }
`;