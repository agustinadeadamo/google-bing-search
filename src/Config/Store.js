/**
 * @desc Dependencies
 */
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

/**
 * @desc Middlewares
 */
import Middlewares from './Middlewares';

/**
 * @desc Reducers
 */
import Reducers from 'Reducer/Reducer';

/**
 * @desc Store
 */
const Store = createStore(Reducers, composeWithDevTools(applyMiddleware(...Middlewares)));

export default Store;
