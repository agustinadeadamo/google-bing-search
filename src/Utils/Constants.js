/* Search mode */
export const SearchMode = {
    "GOOGLE" : 'Google',
    "BING" : 'Bing',
    "GOOGLE_BING" : 'Goolge & Bing',
};

export const MessageError = {
    'SERVER_ERROR': 'There was an error in the server',
    'APP_ERROR': 'There was an error in the app',
};

export const Apikeys = {
    'BING_KEY' : '1ef3364ab9ce48b1b180b109e6aab6fb',
    'GOOGLE_KEY': 'AIzaSyB5Mqgm1gE_sAvmQToGkD2fGnOw-Pl1m5w',
    'GOOGLE_CX': '18ca2e163998cf02d'
}

export const Endpoints = {
    'GOOGLE_SEARCH' : 'https://www.googleapis.com/customsearch/v1',
    'BING_SEARCH': 'https://api.bing.microsoft.com/v7.0/search'
}

