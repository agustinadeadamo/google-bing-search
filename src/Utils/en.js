export const English = {
    'ADD_VALUE_TO_SEARCH' : 'Add a value to search',
    'SELECT_OPTION' : 'Select an option',
    'SUBMIT' : 'Submit',
    'GOOGLE_BING_SEARCH' : 'Google & Bing Search',
}