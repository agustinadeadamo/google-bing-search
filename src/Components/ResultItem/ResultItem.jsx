/**
 * @desc Dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * @desc Styled Component
 */
import { ResultItemComponent } from './style';

const ResultItem = (props) => {

    // Destructuring props
    let {
        title,
        link,
        url,
        snippet
    } = props;

    return (
        <ResultItemComponent data-testid="result">
            <p data-testid="link" className="link">{link}</p>
            <a data-testid="title" className="title" href={url} >{title}</a>
            <p data-testid="snippet" className="snippet">{snippet}</p>
        </ResultItemComponent>
    )

}

ResultItem.propTypes = {
    title: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    snippet: PropTypes.string.isRequired,
  };
  
ResultItem.defaultProps = {
    title: '',
    link: '',
    url: '',
    snippet: '',
};

export default ResultItem;