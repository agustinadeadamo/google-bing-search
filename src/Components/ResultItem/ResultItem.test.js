/**
 * @desc Dependencies
 */
import { create } from 'react-test-renderer';
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux'

/**
 * @desc Config
 */
import Store from 'Config/Store';

/**
 * @desc Utils
 */
import { MessageError } from 'Utils/Constants';

/**
 * @desc Components
 */
import ResultItem from './ResultItem';

describe('[Result Component]',() => {

    it('Renders correctly', () => {

        const {queryByTestId} = render(
            <Provider store={Store}>
                <ResultItem />
            </Provider>
        );

        expect(queryByTestId('result')).toBeTruthy();
    
    });

    it('Matches snapshot', () => {

        const result = create(
            <Provider store={Store}>
                <ResultItem/>
            </Provider>
        ).toJSON;

        expect(result).toMatchSnapshot();
    
    });

    it('Shows correct content', () => {

        // Define values
        let link = 'link test'
        let title = 'title test'
        let snippet = 'snippet test'

        const {getByTestId } = render(
            <Provider store={Store}>
                <ResultItem link={link} title={title} snippet={snippet} message={MessageError.APP_ERROR}/>
            </Provider>
        );
        
        expect(getByTestId('link')).toHaveTextContent(link);
        expect(getByTestId('title')).toHaveTextContent(title);
        expect(getByTestId('snippet')).toHaveTextContent(snippet);

    });

});