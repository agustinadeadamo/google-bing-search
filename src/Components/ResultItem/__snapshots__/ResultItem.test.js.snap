// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Result Component] Matches snapshot 1`] = `[Function]`;
